### ISSUE TITLE
Poor performance using dpm-solver

### ISSUE THREAD

**Issue by RoboticsZhang** [WuJunde/MedSegDiff#134]
After trainning for 17W steps, I can get good results without using dpm-solver in 1000 steps, as shown below:
![image](https://github.com/WuJunde/MedSegDiff/assets/61099841/19a66838-54c2-4714-98e7-2871d75741ef)
However, when using dpm-solver in 20, 50, or 1000 steps, the results all look like the image below:
![image](https://github.com/WuJunde/MedSegDiff/assets/61099841/937d6025-9c32-406d-90af-235d64d7abdc)
I wonder why dpm-solver generate so weird results.
Looking forward to your reply, thank you!

**Comment by ipwjincheng** [comment]
hello,I  met the same question as you,have you solved it?

**Comment by RoboticsZhang** [comment]
> hello,I met the same question as you,have you solved it?

Yep, maybe there are some flaws in the implementation of DPM-solver in MedSegDiff. I used the official version and then solved it. It is worth noting that the official version takes the epsilon as the input, not the mean and covariance, which are the direct output of the neural network.

**Comment by RoboticsZhang** [comment]
> hello,I met the same question as you,have you solved it?

Btw, I plan to open source a customized mini demo with DPM-Solver V2 and Cifar-10 recently. The code architecture follows openai/improved-diffusion, which is the same as MedSegDiff.

**Comment by RoboticsZhang** [comment]
> hello,I met the same question as you,have you solved it?

I  just found that the latest DPM-Solver-V3 have included a demo with openai/guided-diffusion. The link is https://github.com/thu-ml/DPM-Solver-v3 which could help you.

**Comment by zxk72** [comment]
How do you use the official version? Do you need to modify it?

**Comment by zxk72** [comment]
@RoboticsZhang Hello, how to use the official dpm-solver code? How did you modify dpm-solver to make the code run successfully? Is it convenient to publish your modified code? Or can I add you on WeChat? Looking forward to your reply, thank you very much!

**Comment by mumu0310** [comment]
> > 你好，我遇到了和你一样的问题，你解决了吗？
> 
> 


> > 你好，我遇到了和你一样的问题，你解决了吗？
> 
> 我刚刚发现最新的 DPM-Solver-V3 包含一个带有 openai/guided-diffusion 的演示。该链接 https://github.com/thu-ml/DPM-Solver-v3 可以帮助您。

Thank you very much for your help. Have you considered sharing your solution? Or would you like to contact via email?

**Comment by wk19990213** [comment]
> > hello,I met the same question as you,have you solved it?
> 
> I just found that the latest DPM-Solver-V3 have included a demo with openai/guided-diffusion. The link is https://github.com/thu-ml/DPM-Solver-v3 which could help you.

++++兄弟，不会改dpm_solver.py，gaussian_diffusion.py这俩文件呀，你能把这俩文件放你的仓库里吗，疯狂星期四我V你50！！！

**Comment by wk19990213** [comment]
> ++++兄弟，不会改dpm_solver.py，gaussian_diffusion.py这俩文件呀，你能把这俩文件放你的仓库里吗，疯狂星期四我V你50！！！
